src/messages/MOSDOp.h

```diff
@@ -401,6 +417,16 @@ struct ceph_osd_request_head {
       ::decode(osdmap_epoch, p);
       ::decode(flags, p);
       ::decode(reqid, p);
+      ::decode(qos_params, p);
```

Comment:
nit, could just conditionalize this instead of repeating the whole decode logic.

Reply:
I changed as you mentioned. Thank you for reviewing.